### PR TITLE
revise dockerfile for chempropv2

### PR DESCRIPTION
**Pull request by JacksonBurns**
- note that this is locked on a specific version of pytorch, pytorch scatter, and chemprop (down to the commit hash)
 - this only runs on CPU, but includes some minimal instructions to help people get going on GPU
 - tested locally by building, running, and then training a quick FreeSolv model without issue

I have also generally overhauled the Dockerfile based on how we distribute RMG-Py via Docker, see [that Dockerfile](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/main/Dockerfile), which has been working without issues for a while now.

### CODE REVIEW

**Comment by kevingreenman** [comment]
Does this still work without the chemprop commit hash specified, as long as the torch and torch scatter versions are specified? If it works on the latest commit, is there a way we can add this to our CI to be notified when one of our changes breaks this?

**Comment by JacksonBurns** [comment]
Right now this only works with the given commit because we have not yet published rc 1. Once we have that out, we can switch this to instead use whatever the latest release it. Assuming that future versions do not fail the CI, then the docker should work with them. In theory, nothing could break the Docker container that would not also break the CI (both are hardcoding dependency versions RN).

**Comment by JacksonBurns** [comment]
Most robust solution is to switch away from the outdated scatter backend so we can remove this weird installation approach, and then both the CI and the Docker container can just download whatever the latest is.

File: Dockerfile
Comment (davidegraff) [inline Dockerfile:76]: Can we use `ARG`s here, i.e.
```suggestion
# Follow the installation instructions (but with fixed versions, for stability of this image) then clear the cache
RUN python -m pip install torch==${TORCH} --index-url https://download.pytorch.org/whl/${CUDA} && \
    python -m pip install torch-scatter -f https://data.pyg.org/whl/torch-${TORCH}+${CUDA}.html && \
    python -m pip install git+https://github.com/chemprop/chemprop.git@f39a672d003dd82f1fddff8009d98a0c4f21796b && \
    python -m pip cache purge
```

**Review by davidegraff — COMMENTED** [review]
If we're doing everything using `miniconda`, is it not possible to pull from [`continuumio/miniconda3`](https://hub.docker.com/r/continuumio/miniconda3) to avoid the overhead of installing it?

File: Dockerfile
Comment (JacksonBurns) [reply Dockerfile:64]: Feel free to work this out and push it, I think actually running the image with GPU dependencies included will require setting up https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html which I haven't done and I think only works on debian linux.

Comment (davidegraff) [reply Dockerfile:64]: yeah good point- I'm no expert on docker and I don't think it's worth it to spend time figuring this out (at least in the near term). I only suggest this to clearly demarcate where we (or others) will need to go looking in the future for what to edit in the file (and ensure that there's only one source of truth for version info).

**Comment by JacksonBurns** [comment]
@kevingreenman I've rebased so this can be merged into `v2/dev`, unless you want to include the docs updates as part of this PR. If so, we should probably include the stuff about trying to get this to work on GPU wherever that ends up. Could be a good fit for a possible later documentation update PR, since I imagine we might end up with a lot of docs updates for the next rc?

**Review by kevingreenman — CHANGES_REQUESTED** [review]
IMO the best thing to do for documentation is to do it as we go, so everyone who adds something new or changes something should be responsible for updating the corresponding docs. Otherwise, we'll have to trust ourselves to remember everything that needs to be documented later (and inevitably forget some things... \*cough\* `--crossval_index_file` \*cough\*).

An alternative is opening up a separate (draft) PR where we write documentation in parallel, and others contribute to that PR in parallel to their own PRs, but we don't merge it until the next release. 

Or we could open an issue with a checklist that gets continuously updated as we come across more things that need to be documented before the next release (and we do the actual writing later).

In the past, we've (usually) used the 1st approach with chemprop, requiring people to add things to the README when they add new features or make changes. I think this is the cleanest way to do it for v2 as well, just with adding things to the read the docs source files instead of the README.

For this PR, I suggest adding the CPU instructions to the docs as part of this PR and opening a separate issue to remind us that we need to figure out how we should tell people to use docker with GPUs later.

File: docs/source/installation.rst
Comment (davidegraff) [inline docs/source/installation.rst:80]:
```suggestion
Note that not all versions of Chemprop are available as pre-built images, so see the `Docker Hub`_ page for a list of those that are available.
```

File: docs/source/installation.rst
Comment (davidegraff) [inline docs/source/installation.rst:79]:
```suggestion
Where `X`, `Y`, `Z`, and `N`, should be replaced with the version of Chemprop you wish to `pull`. For example, to pull chemprop-2.0.0-rc1, you would run: ``docker pull chemprop/chemprop:2.0.0.rc1``.
```

File: docs/source/installation.rst
Comment (davidegraff) [inline docs/source/installation.rst:87]:
```suggestion
First follow the instructions in `Option 2: Installing from Source`_ up to invoking ``pip`` and then run the following:
```

**Comment by JacksonBurns** [comment]
Docs updates are completed (thanks for the suggestions @davidegraff) and the image has been hand-verified and pushed to Dockerhub. Please re-review @kevingreenman

File: docs/source/installation.rst
Comment (davidegraff) [inline docs/source/installation.rst:80]:
```suggestion
Where ``X``, ``Y``, ``Z``, and ``N``, should be replaced with the version of Chemprop you wish to ``pull``.
For example, to pull ``chemprop-2.0.0rc1`` run
```

File: docs/source/installation.rst
Comment (davidegraff) [inline docs/source/installation.rst:14]:
```suggestion
    We also plan to make chemprop installable using an ``environment.yml`` file with ``conda`` before the release of v2.0.0.
```

**Review by davidegraff — APPROVED** [review]
I hate RST solely because of the need for double-tick monospace formatting.

**Comment by JacksonBurns** [comment]
> I hate RST solely because of the need for double-tick monospace formatting.

I `pandoc` my markdown documents to other formats for this very reason

**Comment by davidegraff** [comment]
> > I hate RST solely because of the need for double-tick monospace formatting.
> 
> I `pandoc` my markdown documents to other formats for this very reason

I feel like it must be possible to write our docs in markdown and compile them into RST at deploy-time. Do you know if something like that is possible?

**Comment by JacksonBurns** [comment]
> I feel like it must be possible to write our docs in markdown and compile them into RST at deploy-time. Do you know if something like that is possible?

It is, but at this point I'm eager to succumb to the sunk-cost fallacy and doom future developers to the plague of RST

**Comment by davidegraff** [comment]
> > I feel like it must be possible to write our docs in markdown and compile them into RST at deploy-time. Do you know if something like that is possible?
> 
> It is, but at this point I'm eager to succumb to the sunk-cost fallacy and doom future developers to the plague of RST

Chemprop V3 but it's just the docs being updated

**Review by kevingreenman — APPROVED** [review]

File: docs/source/installation.rst
Comment (JacksonBurns) [inline docs/source/installation.rst:70]: @kevingreenman if you want to capitalize the link to Docker I think you will have to do it like this

Comment (JacksonBurns) [reply docs/source/installation.rst:70]: Specifically line 70 in this comment

**Pull request merged**
